### PR TITLE
optimize unstructured convert methods

### PR DESCRIPTION
**Pull request by weilaaa**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Instead of adding a conversion function for each resource, we should replace it with a generic conversion function.

### CODE REVIEW

**Comment by RainbowMango** [comment]
Generally looks good. We didn't use the `Golang generic feature` in this patch, right? (I didn't see it, just a confirmation).

**Comment by weilaaa** [comment]
> Generally looks good. We didn't use the `Golang generic feature` in this patch, right? (I didn't see it, just a confirmation).

Not yet, when upgrading to go 1.18, consider using generics.

**Comment by RainbowMango** [comment]
> Not yet, when upgrading to go 1.18, consider using generics.

I'm hesitant to introduce generics even though we are already in Go1.18. Because technically we can use generics in master(v1.3), but can't cherry-pick patches to release branches(<=release-1.2).
Basically, I agree with [Kubernetes Generics Policy](https://github.com/kubernetes/community/blob/master/sig-architecture/generics.md#generics-policy).

**Comment by weilaaa** [comment]
> > Not yet, when upgrading to go 1.18, consider using generics.
> 
> I'm hesitant to introduce generics even though we are already in Go1.18. Because technically we can use generics in master(v1.3), but can't cherry-pick patches to release branches(<=release-1.2). Basically, I agree with [Kubernetes Generics Policy](https://github.com/kubernetes/community/blob/master/sig-architecture/generics.md#generics-policy).

Thanks for sharing, I read k8s generics policy and I agreed. We need to use generics sparingly.

File: pkg/dependenciesdistributor/dependencies_distributor.go [hunk 1]
```diff
@@ -235,24 +235,15 @@ func dependentObjectReferenceMatches(objectKey keys.ClusterWideKey, referenceBin
 }
 
 // OnResourceBindingUpdate handles object update event and push the object to queue.
+// todo:(weilaaa) if should care about error occurs in event handler?
```
Comment (RainbowMango) [inline pkg/dependenciesdistributor/dependencies_distributor.go:238]: We can't do anything except log the error here.
Yeah, please add a log.

Comment (RainbowMango) [reply pkg/dependenciesdistributor/dependencies_distributor.go:238]: Same as below.

**Review by RainbowMango — COMMENTED** [review]
Looks great! 
Let's fix the new added todo with this patch.

**Comment by weilaaa** [comment]
> /lgtm
> 
> You might need to squash your commits.

PTAL again

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**